include/cudaq/Support/TargetConfig.h

```diff
@@ -0,0 +1,190 @@
+/****************************************************************-*- C++ -*-****
+ * Copyright (c) 2022 - 2024 NVIDIA Corporation & Affiliates.                  *
+ * All rights reserved.                                                        *
+ *                                                                             *
+ * This source code and the accompanying materials are made available under    *
+ * the terms of the Apache License 2.0 which accompanies this distribution.    *
+ ******************************************************************************/
+
+#pragma once
+
+#include "llvm/Support/YAMLTraits.h"
+#include <optional>
+#include <string>
+#include <vector>
+
+namespace cudaq {
+namespace config {
+/// Flag to enable feature(s) of the unified NVIDIA target.
+// Use bitset so that we can combine different options, e.g., multi-gpu with
+// fp32/64.
+enum TargetFeatureFlag : unsigned {
```

Comment:
Recommend using scoped enums here ("enum class") if possible (to avoid enum names leaking into the parent namespace). If that makes some of the bit arithmetic more complicated, then feel free to disregard this comment.

Reply:
Yea, this would be tricky to convert to an `enum class`. I added a code comment for this fact.